### PR TITLE
Feat/aag 2854 add request retry

### PR DESCRIPTION
**Pull request by thomas-james-1986**
## JIRA
[AAG-2854]

## DESCRIPTION

This PR adds a retry mechanism that retries requests 5 times.

## SCREENSHOTS
n/a

[AAG-2854]: https://superawesomeltd.atlassian.net/browse/AAG-2854?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: superawesome-base/src/test/java/tv/superawesome/lib/sanetwork/request/TestSANetwork.java [hunk 1]
```diff
@@ -347,5 +383,102 @@ public void test_SANetwork_SendGET_WithDisconnectAtStart () throws Exception {
...
+            } else {
+                assertFalse(success);
+            }
+        });
+
+        //then
+        RecordedRequest record = server.takeRequest();
+        int numberOfRequests = server.getRequestCount();
+        assertEquals("GET /some/url HTTP/1.1", record.getRequestLine());
+        assertEquals(5, numberOfRequests);
```
Comment (me2k) [inline superawesome-base/src/test/java/tv/superawesome/lib/sanetwork/request/TestSANetwork.java:482]: Can we add a timeout test that then succeeds? I realise may take a long time to run, but we could shorten the timeout to make it faster with a retry count of 2 maybe?

**Review by me2k — CHANGES_REQUESTED** [review]
Great work! You've made it look so easy 😁 I'm requesting a change for an additional timeout test so that we can verify things are working properly.

File: superawesome-base/src/test/java/tv/superawesome/lib/sanetwork/request/TestSANetwork.java (hunk 1 above)
Comment (thomas-james-1986) [reply superawesome-base/src/test/java/tv/superawesome/lib/sanetwork/request/TestSANetwork.java:482]: Sure, good idea. I think the timeout was already low enough at 1 second - I've added a test that exceeds it on the first request (1500ms) then succeeds on the second.

File: superawesome-base/src/test/java/tv/superawesome/lib/sanetwork/request/TestSANetwork.java [hunk 2]
```diff
@@ -347,5 +383,136 @@ public void test_SANetwork_SendGET_WithDisconnectAtStart () throws Exception {
...
+                .setResponseCode(200)
+                .setBody(responseBody);
+
+        // when
+        server.enqueue(mockTimeoutResponse);
+        server.enqueue(mockSuccessResponse);
+
+        network.sendGET(url, null, null, (status, payload, success) -> {
+            // then
+            if(status == 200) {
```
Comment (me2k) [inline superawesome-base/src/test/java/tv/superawesome/lib/sanetwork/request/TestSANetwork.java:503]:
```suggestion
            if (status == 200) {
```

**Review by me2k — APPROVED** [review]
Great job 🎉 Thank you for adding that extra test 😁

**Pull request merged**